facades/meal_add_food_facade.js

```diff
@@ -0,0 +1,61 @@
+var Food = require('../models').Food;
+var Meal = require('../models').Meal;
+var MealFood = require('../models').MealFood;
+var pry = require('pryjs')
+
+module.exports = class MealAddFoodFacade {
+
+  static addFood(params) {
+    return new Promise(function(resolve, reject) {
+      findMealAndFood(params)
+      .then(mealAndFood => {
+        handleResponse(mealAndFood)
+        .then(response => {resolve(response)})
+        .catch(error => {reject(error)})
+      })
+      .catch(() => {})
+    })
+  }
+}
+
+function handleResponse(object) {
+  return new Promise(function(resolve, reject){
+    if (object.meal && object.food) {
+      MealFood.create({
+        MealId: object.meal.id,
+        FoodId: object.food.id
+      })
+      .then(mealFood => {
+        if (mealFood) {
+          resolve({status: 201, body: {message: `Successfully added ${object.food.name} to ${object.meal.name}`}})
+        } else {
+          reject({status: 500, body: {error: 'Food was not added to Meal.'}})
+        }
+      })
+      .catch(() => {})
+    } else if (object.meal) {
+      reject({status: 404, body: {error: 'Food not found.'}})
+    } else if (object.food) {
+      reject({status: 404, body: {error: 'Meal not found.'}})
+    } else {
+      reject({status: 404, body: {error: 'Meal and Food not found.'}})
+    }
+  })
+}
+
+function findMealAndFood(params) {
```

Comment:
We might be able to speed this up using a promise.all and performing both lookups simultaneously, it might also reduce the length of the conditional evaluations and lookup methods.